openpype/hosts/maya/plugins/publish/extract_arnold_scene_source.py

```diff
@@ -113,34 +111,62 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
     def _extract(self, nodes, attribute_data, kwargs):
-        self.log.info("Writing: " + kwargs["filename"])
+        self.log.info(
+            "Writing {} with:\n{}".format(kwargs["filename"], kwargs)
+        )
         filenames = []
         # Duplicating nodes so they are direct children of the world. This
         # makes the hierarchy of any exported ass file the same.
-        with delete_after() as delete_bin:
+        with lib.delete_after() as delete_bin:
             duplicate_nodes = []
             for node in nodes:
+                # Only interested in transforms:
+                if cmds.nodeType(node) != "transform":
+                    continue
+
+                # Only interested in transforms with shapes.
+                shapes = cmds.listRelatives(
+                    node, shapes=True, fullPath=True
+                ) or []
+                if not shapes:
+                    continue
+
+                parent = cmds.listRelatives(
+                    node, parent=True, fullPath=True
+                )[0]
                 duplicate_transform = cmds.duplicate(node)[0]
-
-                # Discard the children.
-                shapes = cmds.listRelatives(duplicate_transform, shapes=True)
-                children = cmds.listRelatives(
-                    duplicate_transform, children=True
+                duplicate_transform = "{}|{}".format(
+                    parent, duplicate_transform
                 )
-                cmds.delete(set(children) - set(shapes))
 
                 duplicate_transform = cmds.parent(
                     duplicate_transform, world=True
                 )[0]
 
-                cmds.rename(duplicate_transform, node.split("|")[-1])
-                duplicate_transform = "|" + node.split("|")[-1]
+                basename = node.split("|")[-1].split(":")[-1]
+                duplicate_transform = cmds.rename(
+                    duplicate_transform, basename
+                )
```

Comment:
Since we're forcing it to all be root nodes this will not result in the exact same names if the original hierarchy had non-unique names.

Reply:
We are validating against same named nodes [here](https://github.com/tokejepsen/pype/blob/feature/OP-4859_cant-assign-shaders-to-the-ass-file/openpype/hosts/maya/plugins/publish/validate_arnold_scene_source.py#L47-L51)